src/shared/mount-util.c

```diff
@@ -23,6 +23,58 @@
 #include "string-util.h"
 #include "strv.h"
 
+int mount_fd(const char *source,
+             int target_fd,
+             const char *filesystemtype,
+             unsigned long mountflags,
+             const void *data) {
+
+        char path[STRLEN("/proc/self/fd/") + DECIMAL_STR_MAX(int)];
+
+        xsprintf(path, "/proc/self/fd/%i", target_fd);
+        if (mount(source, path, filesystemtype, mountflags, data) < 0) {
```

Comment:
Hmm, will this show up as /proc/self/fd/nnn in /proc/mounts?

Reply:
It seems it doesn't. Both tables look the same as before.

Reply:
yes, kernel is smart enough for this. Also note, plenty other code is doing this, so there's plenty prior art that relies on this